### ISSUE TITLE
Behavior of QuantizeLinear when input is NaN

### ISSUE THREAD

**Issue by mgehre-amd** [onnx/onnx#5879]
# Ask a Question

### Question
I don't see an explicit specification of the behavior `QuantizeLinear` when going from a floating point type to an integer type and the input value is NaN.
Is the intention to leave the outcome undefined, unspecified, implementation-defined, or is a specific behavior intended?

It's important to understand the intend to build compliant runtimes/compilers.
Thank you!

### Further information
Looking at the onnxruntime implementation
https://github.com/microsoft/onnxruntime/blob/c8ce83967e5b52062558046d769f3af7d871e893/onnxruntime/core/mlas/lib/quantize.cpp#L759
it seems to use a float to integer C++ cast, which has undefined behavior when the float is NaN (see
https://stackoverflow.com/a/10366541/20356323; all computations before that cast in the function just forward the NaN).

**Comment by gramalingam** [comment]
Agree, it would be a good idea to document this. Seems worth discussing. My guess (preference) is "undefined".
